src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -115,4 +115,9 @@ private function getClass($value)
 
         return ltrim(is_object($value) ? get_class($value) : $value, '\\');
     }
+
+    protected function createMetadataClass($class)
```

Comment:
Sorry can you revert to old way as It become more complicated (I was thinking that can allow user to apply custom logic for a class name).